File: lib/models/role.dart
class Role {
  final int roleId;
  final String roleName;
  final String? description;

  Role({
    required this.roleId,
    required this.roleName,
    this.description,
  });

  factory Role.fromJson(Map<String, dynamic> json) {
    int parseInt(dynamic value) {
      if (value is num) {
        return value.toInt();
      }
      return int.tryParse(value?.toString() ?? '') ?? 0;
    }

    return Role(
      roleId: parseInt(json['role_id'] ?? json['id']),
      roleName: (json['role_name'] ?? json['name'] ?? '').toString(),
      description: json['description']?.toString(),
    );
  }
}

File: app/models/role.py
from sqlalchemy import Integer, String
from sqlalchemy.orm import Mapped, mapped_column, relationship

from app.models.base import Base


class Role(Base):
    __tablename__ = "role"

    id: Mapped[int] = mapped_column("role_id", Integer, primary_key=True, index=True)
    role_name: Mapped[str] = mapped_column(String(50), unique=True, nullable=False, index=True)
    description: Mapped[str | None] = mapped_column(String(255), nullable=True)

    users: Mapped[list["User"]] = relationship("User", back_populates="role")
    role_permissions: Mapped[list["RolePermission"]] = relationship(
        "RolePermission",
        back_populates="role",
        cascade="all, delete-orphan",
    )

    @property
    def name(self) -> str:
        return self.role_name
